dartsim/src/EntityManagementFeatures.cc

```diff
@@ -636,7 +694,31 @@ Identity EntityManagementFeatures::ConstructEmptyModel(
         dart::dynamics::Frame::World(),
         _name + "_frame");
 
-  auto [modelID, modelInfo] = this->AddModel({model, modelFrame, ""}, _worldID); // NOLINT
+  auto [modelID, modelInfo] =
+      this->AddModel({model, _name, modelFrame, ""}, _worldID);  // NOLINT
+
+  return this->GenerateIdentity(modelID, this->models.at(modelID));
+}
+
+/////////////////////////////////////////////////
+Identity EntityManagementFeatures::ConstructEmptyNestedModel(
+    const Identity &_modelID, const std::string &_name)
+{
+  // find the world assocated with the model
+  auto worldID = this->models.idToContainerID.at(_modelID);
+  const auto &skel = this->models.at(_modelID)->model;
+  const std::string modelFullName = ::sdf::JoinName(skel->getName(), _name);
+
+  dart::dynamics::SkeletonPtr model =
+      dart::dynamics::Skeleton::create(modelFullName);
+
+  dart::dynamics::SimpleFramePtr modelFrame =
+      dart::dynamics::SimpleFrame::createShared(
+        dart::dynamics::Frame::World(),
+        modelFullName + "_frame");
+
+  auto [modelID, modelInfo] =
+      this->AddModel({model, _name, modelFrame, ""}, worldID);  // NOLINT
```

Comment:
```suggestion
  auto [modelID, modelInfo] =
      this->AddNestedModel({model, _name, modelFrame, ""}, worldID);  // NOLINT
```
Did you mean to call `AddNestedModel` here, since this is a part of the `ConstructEmptyNestedModel` method?

Reply:
Good catch. I changed it to `AddNestedModel` in https://github.com/ignitionrobotics/ign-physics/pull/228/commits/212c941319d4a7227ef38ae6bdbf196f2cec576b. Also in that commit, I added a test expectation that fails without the fix.